core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -562,11 +563,21 @@ trait AttributeComponent {
       }
     }
 
-    private def assertConsistentValueListMembers(attributes: Seq[AttributeValue]): Unit = {
+    private def assertConsistentValueListMembers(attributes: Seq[AttributeValue], attributeName: AttributeName): Unit = {
       if(!attributes.isEmpty) {
         val headAttribute = attributes.head
         if (!attributes.forall(_.getClass == headAttribute.getClass)) {
-          throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"inconsistent attributes for list: $attributes"))
+          // determine the different attribute types
+          val typeNames = attributes.map(_.getClass.getSimpleName).distinct
+          // generate example values for those types
+          val exampleValues = typeNames.flatMap { typeName =>
+            attributes.find(x => x.getClass.getSimpleName == typeName).map(_.toString)
+          }
+          // generate a helpful error message
+          val errMsg = s"inconsistent attributes for list: attribute lists must consist of a single data type. For attribute " +
+            s"'${toDelimitedName(attributeName)}', found types: [${typeNames.mkString(", ")}]. " +
+            s"Sample values for these types: [${exampleValues.mkString(", ")}]"
```

Comment:
Users have been known to place very long lists of `gs://` paths in an `Array`, might want to set a 100-char limit on the length of each element in the examples list.

( did not attempt compiling )
```suggestion
            s"Sample values for these types: [${exampleValues.map(_.take(100)).mkString(", ")}]"
```